pkg/controller/kabaneroplatform/codereadyworkspaces.go

```diff
@@ -67,13 +69,27 @@ func reconcileCRW(ctx context.Context, k *kabanerov1alpha2.Kabanero, c client.Cl
 
 	templateCtx := unstructured.Unstructured{}.Object
 
-	// Deploy the cluster role with the required permissions for codewind.
-	err = processCRWYaml(ctx, k, rev, templateCtx, c, crwYamlNameCodewindClusterRole, true)
+	// Deploy the Codewind cluster role with the required permissions for codewind.
+	err = processCRWYaml(ctx, k, rev, templateCtx, c, crwYamlNameCodewindClusterRole, true, k.GetNamespace())
 	if err != nil {
 		logger.Error(err, fmt.Sprintf("Failed to Apply clusterRole resource. Revision: %v. TemplateCtx: %v", rev, templateCtx))
 		return err
 	}
 
+	// Deploy the Codewind Tekton role
+	err = processCRWYaml(ctx, k, rev, templateCtx, c, crwYamlNameCodewindTektonRole, true, "tekton-pipelines")
```

Comment:
The things that are created in a namespace (in this case the `tekton-pipelines` namespace) can't have an owner reference back to an object in another namespace (in this case the Kabanero CR instance in the `kabanero` namespace).  This means that we'll need cleanup logic to delete the role when either Codewind is disbaled, or when the Kabanero CR instance is deleted.  I think the `cleanupCRW` function does this already.  It calls `deleteCRWOperatorResources()` which deletes the cluster role.  So I think you'd just have to additionally delete this one (and the one on line 87 which is also in `tekton-pipelines`.

Reply:
@kaczyns Thanks Tim, I'll take a look tomorrow.